backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/PreInlineLowering.kt

```diff
@@ -38,6 +49,10 @@ internal class PreInlineLowering(val context: Context) : BodyLoweringPass {
                 if (!enableAssertions && expression.symbol in asserts) {
                     assert(expression.type.isUnit())
                     return IrCompositeImpl(expression.startOffset, expression.endOffset, expression.type)
+                } else if (expression.symbol.owner.isTypeOfIntrinsic()) {
+                    val type = expression.getTypeArgument(0)
+                            ?: org.jetbrains.kotlin.backend.konan.error(container.file, expression, "missing type argument")
+                    return with (kTypeGenerator) { builder.at(expression).irKType(type, needExactTypeParameters = true) }
```

Comment:
But how does typeOf substitute reified type parameters now, if it gets lowered before inline?

Reply:
I mean:
```kotlin
inline fun <reified T> t() = kotlin.reflect.typeOf<T>()

fun main() {
    println(t<String>())
}
```
prints String on JVM. Does this work with Native on this branch?

Reply:
My bad. The reason why I moved it before the inliner was non-passing test, where `reified` parameters aren't expected to be substituted at some places, I'll rethink, what is going on there.